src/backend/cdb/cdbutil.c

```diff
@@ -662,6 +662,9 @@ cdbcomponent_cleanupIdleQEs(bool includeWriter)
 		}
 	}
 
+	/* reset flag in libpq, that is used to avoid stuck in a loop in PQcancel*/
+	PQbypassConnCloseAtCancel(false);
+
```

Comment:
Why are we resetting the flag here? why not in `internal_cancel` immediately after it was used?